packages/@azure/eventhubs/client/lib/connectionContext.ts

```diff
@@ -116,6 +116,10 @@ export namespace ConnectionContext {
         numSenders: Object.keys(connectionContext.senders).length,
         numReceivers: Object.keys(connectionContext.receivers).length
       };
+
+      // Remove all the links and sessions for this AMQP connection.
+       connectionContext.connection.removeAllSessions();
```

Comment:
We should also call `close()` on the cbsSession and managementSession to ensure that the event listeners added in rhea-promise for these get removed too.

For normal sender/receiver, this will be taken care of by the _closeLink() call